xbmc/interfaces/legacy/Dialog.h

```diff
@@ -679,6 +680,131 @@ namespace XBMCAddon
       bool iscanceled();
 #endif
     };
+
+    //@}
+
+    ///
+    /// \defgroup python_DialogBusy DialogBusy
+    /// \ingroup python_xbmcgui
+    /// @{
+    /// @brief <b>Kodi's progress dialog class (Duh!)</b>
+    ///
+    ///
+    class DialogBusy : public AddonClass
+    {
+      CGUIDialogBusy* dlg;
+      bool open;
+
+    protected:
+      virtual void deallocating();
+
+    public:
+
+      DialogBusy() : dlg(NULL), open(false) {}
+      virtual ~DialogBusy();
+
+#ifdef DOXYGEN_SHOULD_USE_THIS
+      ///
+      /// \ingroup python_DialogBusy
+      /// \python_func{ xbmcgui.DialogBusy().create() }
+      ///------------------------------------------------------------------------
+      ///
+      /// Create and show a busy dialog.
+      ///
+      /// @note Use update() to update the progressbar.
+      ///
+      ///
+      ///------------------------------------------------------------------------
+      ///
+      /// **Example:**
+      /// ~~~~~~~~~~~~~{.py}
+      /// ..
+      /// dialog = xbmcgui.DialogBusy()
+      /// dialog.create()
+      /// ..
+      /// ~~~~~~~~~~~~~
+      ///
+      create(...);
+#else
+      void create();
+#endif
+
+#ifdef DOXYGEN_SHOULD_USE_THIS
+      ///
+      /// \ingroup python_DialogBusy
+      /// \python_func{ xbmcgui.DialogBusy().update(percent) }
+      ///------------------------------------------------------------------------
+      ///
+      /// Updates the busy dialog.
+      ///
+      /// @param percent        integer - percent complete. (0:100)
+      ///
+      /// @note If percent == 0, the progressbar will be hidden.
```

Comment:
Yup, I just did it in same way as progress dialog. I agree that it makes sense to change this behaviour for both.